pkg/aws/testutil_test.go

```diff
@@ -62,22 +64,52 @@ const (
 	fakeRDSSnapshotName = "rds-snapshot"
 )
 
-func fakeReportItemDeleting() *clusterservice.ReportItem {
-	return &clusterservice.ReportItem{
-		ID:           fakeRDSClientInstanceARN,
-		Name:         fakeRDSClientInstanceIdentifier,
+func mockReportItemDeleted(modifyFn func(*clusterservice.ReportItem)) *clusterservice.ReportItem {
+	mock := &clusterservice.ReportItem{
+		Name:         fakeResourceIdentifier,
+		Action:       clusterservice.ActionDelete,
+		ActionStatus: clusterservice.ActionStatusComplete,
+	}
+	if modifyFn != nil {
+		modifyFn(mock)
+	}
+	return mock
+}
+
+func mockReportItemDeleting(modifyFn func(*clusterservice.ReportItem)) *clusterservice.ReportItem {
+	mock := &clusterservice.ReportItem{
+		Name:         fakeResourceIdentifier,
 		Action:       clusterservice.ActionDelete,
 		ActionStatus: clusterservice.ActionStatusInProgress,
 	}
+	if modifyFn != nil {
+		modifyFn(mock)
+	}
+	return mock
 }
 
-func fakeReportItemDryRun() *clusterservice.ReportItem {
-	return &clusterservice.ReportItem{
-		ID:           fakeRDSClientInstanceARN,
-		Name:         fakeRDSClientInstanceIdentifier,
+func mockReportItemDryRun(modifyFn func(*clusterservice.ReportItem)) *clusterservice.ReportItem {
+	mock := &clusterservice.ReportItem{
+		Name:         fakeResourceIdentifier,
 		Action:       clusterservice.ActionDelete,
 		ActionStatus: clusterservice.ActionStatusDryRun,
 	}
+	if modifyFn != nil {
+		modifyFn(mock)
+	}
+	return mock
+}
```

Comment:
Can we either have a single `func mockReportItem(modifyFn func(*clusterservice.ReportItem)) *clusterservice.ReportItem` function and call that explicitly in the tests

or

have 

```
func mockReportItemDryRun(modifyFn func(*clusterservice.ReportItem)) *clusterservice.ReportItem {
	mock := mockReportItem(func(item *clusterservice.ReportItem) {
               if modifyFn != nil {
                      modifyFn(item)
               }
               item.ActionStatus = clusterservice.ActionStatusDryRun
        })
	return mock
}
```

I'd be for the first